withdraw-by-check/REPWRITERSPECS/BANNO.CHECK.WITHDRAW.V1.POW

```diff
@@ -175,374 +266,357 @@ PRINT TITLE="BANNO.CHECK.WITHDRAW.V1.POW"
+ 
+ IF @RGSTATE=STATESTART THEN
+  DO
+   IF BNOERRORCODE=0 THEN
+    CALL JSONVALIDACCT
+   ELSE 
+    CALL JSONINVALIDACCT
+  END
+ ELSE IF @RGSTATE=STATEPERFORMWITHDRAW THEN
+  DO
+   IF BNOERRORCODE=0 THEN
+    CALL JSONTRANSUCCESS 
+   ELSE
+    DO
+     IF VALIDATESTATUS<>VALIDATEOK THEN
+      CALL JSONINVALIDACCT
+     ELSE
+      CALL JSONTRANFAILED 
+    END
+  END
+
  CALL BNOPRINTPROGRAMINFO
  CALL BNOPRINTRGDATA
  CALL BNOPRINTDEBUG
 
- PRINT "  }"
- NEWLINE
+ CALL JSONCLOSE
 END [PRINT]
 
-PROCEDURE RETURNVALIDACCT
-[* Set the return JSON code for a valid account
+PROCEDURE INITIALIZEDATA
+[* Initialize state variables
 *]
- PRINT "{"
- NEWLINE
- PRINT "  "+Q+"results"+Q+": {"
- NEWLINE
- PRINT "    "+Q+"eligible"+Q+": true,"
- NEWLINE
- PRINT "    "+Q+"memberAccountNumber"+Q+": "+Q+@RGUSERCHR1+Q+","
- NEWLINE
- PRINT "    "+Q+"shareLoanDescription"+Q+": "+Q+SLDESCRIPTION+Q+","
- NEWLINE
- PRINT "    "+Q+"available"+Q+": "+Q+SLAVAILABLEBALCHR+Q+","
- NEWLINE
- TMPCHR=FORMAT("########9.99",PARAMMINWDLIMIT)
- CALL NLTS
- PRINT "    "+Q+"minWdAmount"+Q+": "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=FORMAT("########9.99",PARAMMAXWDLIMIT)
- CALL NLTS
- PRINT "    "+Q+"maxWdAmount"+Q+": "+Q+TMPCHR+Q+","
- NEWLINE
- PRINT "    "+Q+"owner"+Q+": "+Q+NAME:LONGNAME+Q+","
- NEWLINE
- PRINT "    "+Q+"address"+Q+": ["
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:1
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:2
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:3
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:4
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:5
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+","
- NEWLINE
- TMPCHR=ACCOUNT:PAYEELINE:6
- CALL NLTS
- PRINT "      "+Q+TMPCHR+Q+""
- NEWLINE
- PRINT "    ],"
- NEWLINE
- PRINT "    "+Q+"disclaimerText"+Q+": ["
- NEWLINE
+ WHILELIMIT=100000
+ RGMEMBERNUM=""
+ RGSLTYPE=""
+ RGSLID=""
+ RGWDAMOUNT=$0.00
 
- FOR TMPLOOP=1 TO PARAMCUSTOMTERMLINES
+ BNOERRORCODE=0
+ BNODEBUGMODE=FALSE
+ LFNUMBER=0
+ LFERROR=""
+ READCONFIGERROR=FALSE
+ READCONFIGERRORMSG=""
+ LFERROR=""
+ LFNUMBER=0
+ ADDRESSLOC=0
+ NADDRESSTYPE=-1
+ PARAMREGD=TRUE
+ PARAMVALIDSHARETYPES="ALL"
+ PARAMVALIDLOANTYPES="ALL"
+ PARAMCASHADVLIMIT=$0.00
+ PARAMMINWDLIMIT=$0.00
+ PARAMMAXWDLIMIT=$0.00
+ PARAMWARNINGSACCT="NONE"
+ PARAMWARNINGSSHARE="NONE"
+ PARAMWARNINGSLOAN="NONE"
+ PARAMCUSTOMTERMLINES=0
+ PARAMERRDISPLAYCOUNT=0
+ FOR TMPLOOP=0 TO PARAMERRDISPLAYMAX
   DO
-   TMPCHR=PARAMCUSTOMTERMS(TMPLOOP)
-   CALL NTS
-   PRINT "   "
-   PRINT Q
-   IF TMPLOOP>1 AND
-      TMPCHR<>"" AND
-      PARAMCUSTOMTERMS(TMPLOOP-1)<>"" THEN
-    PRINT " "
-   PRINT TMPCHR+Q
-   IF TMPLOOP<PARAMCUSTOMTERMLINES THEN
-    PRINT ","
-   NEWLINE
+   PARAMERRDISPLAY(TMPLOOP,ERRDISPLAYERRCODE)=0
+   PARAMERRDISPLAY(TMPLOOP,ERRDISPLAYLINECOUNT)=0
   END
 
- PRINT "]"
- PRINT "  }"
- NEWLINE
+ VALIDATESTATUS=VALIDATEOK
+ TRANSTATUS=TRANOK
+ 
```

Comment:
The following was removed from INITAILIZEDATA. I think it should be added back:

 FOR TMPLOOP=0 TO PARAMCUSTOMTERMLINESMAX
  DO
   PARAMCUSTOMTERMS(TMPLOOP)=""
  END
 PARAMCUSTOMTERMLINES=0

Reply:
By setting PARAMCUSTOMTERMLINES to zero, you are initializing the array.  I don't see a need to set each element in the array to a blank screen.  We should only be accessing things from 0 to PARAMCUSTOMTERMLINES in the array.  Let me know if you see another reason why that extra code should exist.